File: Homework6/Program/View/Menu.java
package Homework6.Program.View;

import Homework6.Program.View.Base.IView;

public class Menu {
    IView view;
    String[] menu;

    public Menu(IView view, String[] menu){
        this.view = view;
        this.menu = menu;
    }

    public void show() {
        view.printMessage("===========");
        for (int i = 0; i < menu.length; i++) {
            view.printMessage(String.format("%d - %s", i + 1, menu[i]));
        }
        view.printMessage("===========");
    }

    public String getUserChoice(String prompt){
        return view.getUserInput(prompt);
    }
    
    public int count(String[] menu){
        return menu.length;
    }
}

File: Homework7/View/menu.py
import view
class Menu:

    def __init__(self, menu_items: list) -> None:
        self.menu_items = {str(key): val for key, val in enumerate(menu_items, 1)}
        self.view = view.View()

    def show(self):
        for k, v in self.menu_items.items():
            self.view.print_message(f'{k} - {v}')
        return 
    
    def get_user_choice(self):
        while True:
            self.show()
            k = self.view.get_user_input('Ваш выбор: ')
            if k in self.menu_items.keys(): return k
            else: print("Неверный ввод!")